### DISCUSSION TITLE
Examples for using the BPC profile

### DISCUSSION THREAD

**Discussion by chamikaCN** [phax/phase4#85] [Q&A]
Hi, are there any resources or examples for setting up the AS4 with BPC profile available?

**Comment by phax** [comment]
Unfortunately not yet. But I plan to do this next Sunday as a separate "phase4-client-bpc" submodule - so please stay tuned :)

**Reply by chamikaCN** [reply]
Good to hear it, I tried to go through the implementation of Peppol and ENTSOG client modules. As I understood,
- Phase4BPCException
- Phase4BPCHttpClientSettings
- Phase4BPCSender
classes should be implemented in order to make the BPC client? Is that correct?

**Reply by phax** [reply]
yes exactly

**Reply by chamikaCN** [reply]
Should the Validation related Classes and Interfaces be implemented, as you have done for peppol client?

**Reply by phax** [reply]
Well, you could do that, if you consider it stable enough. This is an "added value" to users, but it needs to be maintained.
Then the rules should fit into the structure as provided by phive-rules.
What is your opinion - is it worth the effort? And if so, what would be the source of the validation rules?

**Reply by chamikaCN** [reply]
Hi, I have been working on this. First I have implemented the aforementioned three classes. Then I created `BPCIdentityFactory` and 
- BPCParticipantIdentifier
- BPCDocumentTypeIdentifier
- BPCProcessIdentifier classes as well similar to what you have done in Peppol Client. My Idea is that, this along with certificate validation is sufficient for a basic BPC client module, isn't it?
